rules/CodeQuality/Rector/LogicalAnd/LogicalToBooleanRector.php

```diff
@@ -55,10 +55,22 @@ public function getNodeTypes(): array
      */
     public function refactor(Node $node): ?Node
     {
+        $left = $node->left;
+        if ($left instanceof LogicalOr || $left instanceof LogicalAnd) {
+            /** @var BooleanAnd|BooleanOr $left */
+            $left = $this->refactor($left);
+        }
+
+        $right = $node->right;
+        if ($right instanceof LogicalOr || $right instanceof LogicalAnd) {
+            /** @var BooleanAnd|BooleanOr $right */
+            $right = $this->refactor($right);
+        }
+
         if ($node instanceof LogicalOr) {
-            return new BooleanOr($node->left, $node->right);
+            return new BooleanOr($left, $right);
         }
 
-        return new BooleanAnd($node->left, $node->right);
+        return new BooleanAnd($left, $right);
```

Comment:
this whole code inside refactor can be moved to separate method, eg: processLogicalToBoolean(), so the recursive method is in the new method.

For clean rebase, I created a blogpost about it: https://samsonasik.wordpress.com/2015/09/16/practical-git-4-rebasing-conflicted-task-branch-against-primary-branch/

Reply:
Moved the code, is this what you wanted?